backend/cn/lib/check.ml

```diff
@@ -1509,50 +1482,45 @@ let rec check_expr labels (e : BT.t Mu.expr) (k : IT.t -> unit m) : unit m =
           let@ () = WellTyped.WCT.is_ct act.loc act.ct in
           let@ () = WellTyped.ensure_base_type loc ~expect Bool in
           let@ () = WellTyped.ensure_base_type loc ~expect:(Loc ()) (Mu.bt_of_pexpr pe) in
-          (* TODO (DCM, VIP): check. Also: this is the same as PtrWellAligned *)
+          (* TODO (DCM, VIP): error if called on Void or Function Ctype *)
           check_pexpr pe (fun arg ->
-            let value = aligned_ (arg, act.ct) loc in
-            k value)
+            (* let unspec = CF.Undefined.UB_unspec_pointer_add in *)
+            (* let@ () = check_has_alloc_id loc arg unspec in *)
+            (* let index = num_lit_ Z.one Memory.uintptr_bt here in *)
+            (* let check_this = arrayShift_ ~base:arg ~index act.ct loc in *)
+            (* let ub = CF.Undefined.(UB_CERB004_unspecified unspec) in *)
+            (* let@ () = check_live_alloc_bounds `ISO_array_shift loc ub [ check_this ] in *)
```

Comment:
This is not quite right - the resource check may fail, but the type checking should succeed (and the value returned should be false).